### PR TITLE
Use prettier for all js files

### PR DESCRIPTION
**Pull request by JamesGardiner**
### What is the context of this PR?
Prettier was only being used to format and lint json. This expand it to cover JS files in the functional tests directory and subdirectories.

### How to review 
Check functional tests still pass. Sense check the style changes.

### CODE REVIEW

**Comment by MebinAbraham** [comment]
Change base branch?.

**Comment by JamesGardiner** [comment]
> Change base branch?.

Done 👍

**Comment by MebinAbraham** [comment]
Need to merge master and fix any conflicts.

File: tests/functional/base_pages/hub.page.js [hunk 1]
```diff
@@ -1,47 +1,72 @@
...
+  }
 
-  errorHeader() { return '#main-content > div.panel.panel--error.u-mb-s > div.panel__header > div'; }
+  errorHeader() {
+    return "#main-content > div.panel.panel--error.u-mb-s > div.panel__header > div";
+  }
 
-  errorNumber(number = 1) { return '[data-qa="error-body"] ul > li:nth-child(' + number + ') > a'; }
+  errorNumber(number = 1) {
+    return '[data-qa="error-body"] ul > li:nth-child(' + number + ") > a";
```
[the remark this answers is not in the document]
Comment (JamesGardiner) [reply tests/functional/base_pages/hub.page.js:29]: Good shout - I'm happy to do that in this PR

File: package.json [hunk 2]
```diff
@@ -12,12 +12,15 @@
   },
   "scripts": {
     "start": "make dev-compose-up && concurrently \"make run\" \"livereload . -e 'html,njk'\"",
-    "lint": "yarn lint:tests && yarn lint:test-schemas",
+    "lint": "yarn lint:tests && yarn lint:test-schemas && yarn lint:prettier-check",
+    "lint:prettier-check": "prettier --check tests/functional/**/*.js && prettier --check test_schemas/*/*.json",
     "lint:tests": "rm -rf ./tests/functional/generated_pages && eslint tests/functional/spec --fix",
     "lint:test-schemas": "eslint test_schemas/*/*.json",
```
[the remark this answers is not in the document]
Comment (JamesGardiner) [reply package.json:18]: Good spot - it doesn't need to (and shouldn't) touch generated_pages at that point. I've removed it.

As for the --fix flag, again I agree this shouldn't be fixing the code in-place. I've removed it in favour of throwing an error should it fail the eslint check.

File: package.json [hunk 3]
```diff
@@ -13,11 +13,13 @@
...
     "start": "make dev-compose-up && concurrently \"make run\" \"livereload . -e 'html,njk'\"",
     "lint": "yarn lint:tests && yarn lint:test-schemas",
-    "lint:tests": "rm -rf ./tests/functional/generated_pages && eslint tests/functional/spec --fix",
-    "lint:test-schemas": "eslint test_schemas/*/*.json",
+    "lint:tests": "prettier --check tests/functional/**/*.js && eslint tests/functional/spec",
+    "lint:test-schemas": "prettier --check test_schemas/*/*.json && eslint test_schemas/*/*.json",
     "test_functional": "./node_modules/.bin/wdio tests/functional/wdio.conf.js",
     "generate_pages": "rm -rf ./tests/functional/generated_pages && pipenv run python -m tests.functional.generate_pages schemas/en/ ./tests/functional/generated_pages -r '../../base_pages' && pipenv run python -m tests.functional.generate_pages test_schemas/en/ ./tests/functional/generated_pages -r '../../base_pages'",
-    "format": "prettier test_schemas/*/*.json --write"
+    "format": "yarn format:tests && yarn format:test-schemas",
```
Comment (MebinAbraham) [inline package.json:20]: I think this should include `eslint ... --fix`

File: package.json [hunk 4]
```diff
@@ -13,11 +13,13 @@
...
+    "format:tests": "prettier tests/functional/**/*.js --write",
```
Comment (MebinAbraham) [inline package.json:21]: I think this should also include `eslint ... --fix`

File: package.json [hunk 5]
```diff
@@ -13,19 +13,25 @@
...
     "lint": "yarn lint:tests && yarn lint:test-schemas",
-    "lint:tests": "rm -rf ./tests/functional/generated_pages && eslint tests/functional/spec --fix",
-    "lint:test-schemas": "eslint test_schemas/*/*.json",
+    "lint:tests": "prettier --check tests/functional/**/*.js && eslint tests/functional/spec",
+    "lint:test-schemas": "prettier --check test_schemas/*/*.json && eslint test_schemas/*/*.json",
     "test_functional": "./node_modules/.bin/wdio tests/functional/wdio.conf.js",
     "generate_pages": "rm -rf ./tests/functional/generated_pages && pipenv run python -m tests.functional.generate_pages schemas/en/ ./tests/functional/generated_pages -r '../../base_pages' && pipenv run python -m tests.functional.generate_pages test_schemas/en/ ./tests/functional/generated_pages -r '../../base_pages'",
-    "format": "prettier test_schemas/*/*.json --write"
+    "format": "yarn format:tests && yarn format:test-schemas",
+    "format:tests": "prettier tests/functional/**/*.js --write && eslint --fix tests/functional/**/*.js",
```
Comment (ajmaddaford) [inline package.json:21]: Why does eslint use `tests/functional/**/*.js` here and `tests/functional/spec` in the lint command?

Comment (JamesGardiner) [reply package.json:21]: They should both use /**/*.js - the `/tests/functional/spec` was previously what was targeted, but there is no reason to be that restrictive

**Comment by ajmaddaford** [comment]
@JamesGardiner Looks like a good time to get this in as there are not any pull requests making changes to functional tests.

**Pull request merged**